test/layout/test_teleporters.py

```diff
@@ -100,26 +157,57 @@ def test_data(request):
     )
 
 
-def test_decode(test_data):
+def test_valid_targets(test_echoes_data):
+    game = RandovaniaGame.METROID_PRIME_ECHOES
+    config = PrimeTrilogyTeleporterConfiguration(
+        mode=TeleporterShuffleMode.ONE_WAY_TELEPORTER,
+        skip_final_bosses=False,
+        allow_unvisited_room_names=False,
+        excluded_teleporters=TeleporterList((), game),
+        excluded_targets=TeleporterTargetList((), game),
+    )
+    targets = config.valid_targets
+    assert targets != [test_echoes_data]
```

Comment:
This is wrong. This should be doing an `==` comparison, worst case maybe just a length one.